### ISSUE TITLE
Is there a way to dynamically resolve attributes?

### ISSUE THREAD

**Issue by DarkoreXOR** [nilproject/NiL.JS#200]
I need to catch an event when the script tries to access the attribute of JSObject, but the attribute does not exist yet and create it in this event.

**Comment by nilproject** [comment]
No, at this moment this is impossible. Engine reads attributes in many places of code and should do it as fast as possible. If this is necessary, you can patch JSValue._attributes, replace it with property (so, in this way JSValue.Attributes will be "property for property") and handle this event.
